escrow/state_service.go

```diff
@@ -80,8 +78,44 @@ func (service *PaymentChannelStateService) GetChannelState(context context.Conte
 		return nil, fmt.Errorf("channel is not found, channelId: %v", channelID)
 	}
 
+	//For backward compatibility
+	oldProto := false
+	blockNumberPassed := int64(request.CurrentBlock)
+
+	// signature verification
+	message := bytes.Join([][]byte{
+		[]byte("__get_channel_state"),
+		channelID.Bytes(),
+		abi.U256(big.NewInt(int64(request.CurrentBlock))),
+	}, nil)
+	signature := request.GetSignature()
+
+	sender, err := authutils.GetSignerAddressFromMessage(message, signature)
+	if err != nil {
```

Comment:
Wont the older signature error out?

Reply:
if an invalid signature is sent it will error ( which we want to)  , however valid signatures will be interpreted and they dont error  ( re tested this on old snet-cli and confirmed ) but the sender will not match